WordPress/src/main/java/org/wordpress/android/reader/savedposts/resolver/ReaderSavedPostsResolver.kt

```diff
@@ -0,0 +1,102 @@
+    private val jetpackReaderSavedPostsFlag: JetpackReaderSavedPostsFlag,
+    private val queryResult: QueryResult,
+    private val contextProvider: ContextProvider,
+    private val contentResolverWrapper: ContentResolverWrapper,
+    private val readerSavedPostsAnalyticsTracker: ReaderSavedPostsAnalyticsTracker,
+    private val wordPressPublicData: WordPressPublicData,
+    private val appPrefsWrapper: AppPrefsWrapper,
+    private val readerPostTableWrapper: ReaderPostTableWrapper,
+    private val readerTagTableWrapper: ReaderTagTableWrapper,
+    private val readerDatabaseWrapper: ReaderDatabaseWrapper
+) {
+    fun tryGetReaderSavedPosts(onSuccess: () -> Unit, onFailure: () -> Unit) {
+        val isFeatureFlagEnabled = jetpackReaderSavedPostsFlag.isEnabled()
+        if (!isFeatureFlagEnabled) {
+            onFailure()
+            return
+        }
+        val isFirstTry = appPrefsWrapper.getIsFirstTryReaderSavedPostsJetpack()
+        if (!isFirstTry) {
+            onFailure()
+            return
+        }
+
+        readerSavedPostsAnalyticsTracker.trackStart()
+        appPrefsWrapper.saveIsFirstTryReaderSavedPostsJetpack(false)
+        val savedPostsCursor = getReaderSavedPostsCursor()
+        if (savedPostsCursor != null) {
+            val posts = queryResult.getValue<ReaderPostList>(savedPostsCursor) ?: ReaderPostList()
+
+            if (posts.isNotEmpty()) {
+                updateReaderSavedPosts(onSuccess, onFailure, posts)
+            } else {
+                readerSavedPostsAnalyticsTracker.trackFailed(ErrorType.NoUserSavedPostsError)
+                onFailure()
+            }
+        } else {
+            readerSavedPostsAnalyticsTracker.trackFailed(ErrorType.QuerySavedPostsError)
+            onFailure()
+        }
+    }
+
+    private fun getReaderSavedPostsCursor(): Cursor? {
+        val savedPostsUriValue =
+                "content://${wordPressPublicData.currentPackageId()}.${ReaderSavedPostsProvider::class.simpleName}"
+        return contentResolverWrapper.queryUri(
+                contextProvider.getContext().contentResolver,
+                savedPostsUriValue
+        )
+    }
+
+    @Suppress("TooGenericExceptionCaught", "SwallowedException")
+    private fun updateReaderSavedPosts(
+        onSuccess: () -> Unit,
+        onFailure: () -> Unit,
+        posts: ReaderPostList
+    ) {
+        try {
+            readerDatabaseWrapper.reset(false)
+            readerTagTableWrapper.addOrUpdateTag(
+                    ReaderTag(
+                            "",
+                            contextProvider.getContext().getString(R.string.reader_save_for_later_display_name),
+                            contextProvider.getContext().getString(R.string.reader_save_for_later_title),
+                            "",
+                            BOOKMARKED
+                    )
+            )
+
+            readerTagTableWrapper.getBookmarkTags()!!.first().let {
```

Comment:
Would it make sense to throw an explicit Exception instead of using the non-null assertion operator?
e.g.
```
readerTagTableWrapper.getBookmarkTags()?.let {
            readerPostTableWrapper.addOrUpdatePosts(it.first(), posts)
} ?: throw NullPointerException("${ReaderTagList::class.java.simpleName} not found.")
```

Reply:
Resolving this one since it seems the same as [this one](https://github.com/wordpress-mobile/WordPress-Android/pull/17290#discussion_r992566664). Let me know if I'm wrong of course 🙇

Reply:
It's the same. My bad, my first review was "gone" from GitHub and I thought it wasn't submitted, but apparently it was 🤷‍♂️